### PR TITLE
ci: configure workflows

### PR DESCRIPTION
**Pull request by OrKoN**
- include generated files into git so that the package can be installed from GitHub directly
- pin specs using git submodules (to be rolled)
- adds ci and publish jobs

### CODE REVIEW

File: tools/build.sh [hunk 1]
```diff
@@ -4,23 +4,17 @@
...
-
-npx prettier -w src/**/*.ts
-npx prettier -w test-d/**/*.ts
+cddlconv specs/webdriver-bidi/all.cddl > src/gen/main.ts
+cddlconv specs/permissions/all.cddl > src/gen/permissions.ts
 
 npx tsc -p tsconfig.json
-npx tsd
\ No newline at end of file
+npx tsd
```
Comment (Lightning00Blade) [inline tools/build.sh:19]: Why do we need this `tsd`?

**Review by Lightning00Blade — APPROVED** [review]
LGTM

File: tools/build.sh (hunk 1 above)
Comment (OrKoN) [reply tools/build.sh:19]: testing types

Comment (OrKoN) [reply tools/build.sh:19]: so that we are sure the generator didn't break anything

Comment (Lightning00Blade) [reply tools/build.sh:19]: Aaa `tsc` vs `tsd`. My bad. Thanks

**Pull request merged**